crates/yttrium/src/provider_pool.rs

```diff
@@ -123,34 +126,33 @@ impl ProviderPool {
         path: &str,
         additional_query_params: impl IntoIterator<Item = (&str, &str)>,
     ) -> RootProvider {
-        // let providers = self.providers.read().await;
-        // let cached_provider = providers.get(chain_id);
-        let cached_provider = None as Option<&RootProvider>;
+        let cached_provider =
+            self.eip155_providers.read().await.get(chain_id).cloned();
         if let Some(provider) = cached_provider {
-            provider.clone()
+            provider
         } else {
-            // std::mem::drop(providers);
```

Comment:
Don't need `drop()` anymore because `cached_provider` now owns its value (via `.cloned()`) instead of being a reference to the mutex guard. This would cause deadlocks when trying to insert the provider. But we would clone the value anyway, so this avoids the extra drop.